dist/android/app/src/main/java/com/meetup/swarmicons/DetailActivity.kt

```diff
@@ -0,0 +1,48 @@
+package com.meetup.swarmicons
+
+import android.app.Activity
+import android.content.Context
+import android.content.Intent
+import android.os.Bundle
+import android.view.MenuItem
+import kotlinx.android.synthetic.main.activity_detail.*
+
+
+class DetailActivity : Activity() {
+    val drawableName get() = intent.getStringExtra("drawable_name")!!
+    val drawableId get() = intent.getIntExtra("drawable_id", 0)
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        (intent.hasExtra("drawable_name") && intent.hasExtra("drawable_id")) || throw IllegalArgumentException()
+
+        setContentView(R.layout.activity_detail)
+
+        title = drawableName
+        actionBar.apply {
+            setDisplayHomeAsUpEnabled(true)
+        }
+
+        detailImage.apply {
+            setImageResource(drawableId)
+            transitionName = "icon_$drawableName"
+        }
+
+        window.sharedElementEnterTransition.duration = 200
+    }
+
+    override fun onOptionsItemSelected(item: MenuItem): Boolean =
+        when (item.itemId) {
+            android.R.id.home -> { finish(); true }
+            else -> super.onOptionsItemSelected(item)
+        }
+
+    companion object {
```

Comment:
The other way we could do is the following:

companion object {
        val NAME = "DetailActivity:drawable_name"
        val ID = "DetailActivity:drawable_id"
    }

And where you are calling, instead of:
val opts = ActivityOptions.makeSceneTransitionAnimation(this, view, transitionName)
startActivity(DetailActivity.intent(this, info), opts.toBundle())

You could do:
startActivity(intentFor<DetailActivity>(
                            DetailActivity.NAME to info.first,
                            DetailActivity.ID to info.second), opts.toBundle())

val drawableName get() = intent.getStringExtra(NAME)!!
val drawableId get() = intent.getIntExtra(ID, 0)

Reply:
we don't have (and don't need) anko in this stupid little app

Reply:
hmm, I thought was there, my bad.